public/app/features/admin/UserListAdminPage.tsx

```diff
@@ -0,0 +1,141 @@
+import React, { useEffect } from 'react';
+import { hot } from 'react-hot-loader';
+import { connect, MapDispatchToProps, MapStateToProps } from 'react-redux';
+import { NavModel } from '@grafana/data';
+import { Pagination, Forms, ActionBar, ActionBarSpacing, Tooltip } from '@grafana/ui';
+import { StoreState, UserDTO } from '../../types';
+import Page from 'app/core/components/Page/Page';
+import { getNavModel } from '../../core/selectors/navModel';
+import { fetchUsers, changeQuery, changePage } from './state/actions';
+import { TagBadge } from 'app/core/components/TagFilter/TagBadge';
+
+interface OwnProps {}
+
+interface ConnectedProps {
+  navModel: NavModel;
+  users: UserDTO[];
+  query: string;
+  showPaging: boolean;
+  totalPages: number;
+  page: number;
+}
+
+interface DispatchProps {
+  fetchUsers: typeof fetchUsers;
+  changeQuery: typeof changeQuery;
+  changePage: typeof changePage;
+}
+
+type Props = OwnProps & ConnectedProps & DispatchProps;
+
+const UserListAdminPageUnConnected: React.FC<Props> = props => {
+  useEffect(() => {
+    props.fetchUsers();
+  }, []);
+
+  return (
+    <Page navModel={props.navModel}>
+      <Page.Contents>
+        <>
+          <ActionBar>
+            <Forms.Input
+              size="md"
+              type="text"
+              placeholder="Find user by name/login/email"
+              tabIndex={1}
+              autoFocus={true}
+              value={props.query}
+              spellCheck={false}
+              onChange={event => props.changeQuery(event.currentTarget.value)}
+              prefix={<i className="fa fa-search" />}
+            />
+            <ActionBarSpacing />
+            <Forms.LinkButton href="admin/users/create" variant="primary">
+              New user
+            </Forms.LinkButton>
+          </ActionBar>
+
+          <div className="admin-list-table">
+            <table className="filter-table form-inline filter-table--hover">
+              <thead>
+                <tr>
+                  <th></th>
+                  <th>Login</th>
+                  <th>Email</th>
+                  <th>
+                    Seen&nbsp;
+                    <Tooltip placement="top" content="Time since user was seen using Grafana">
+                      <i className="fa fa-question-circle" />
+                    </Tooltip>
+                  </th>
+                  <th></th>
+                  <th style={{ width: '1%' }}></th>
+                </tr>
+              </thead>
+              <tbody>{props.users.map(renderUser)}</tbody>
+            </table>
+          </div>
+          <Pagination
+            visible={props.showPaging}
```

Comment:
Don't render rather than passing prop maybe?

Reply:
Could be, on the other hand it delegates the conditional render to `Pagination` and just reads better.

Reply:
I guess it is a matter of taste but I will change to @dprokop suggestions since it is more consistent with the rest of the code base.